src/display/text_layer.js

```diff
@@ -528,20 +534,48 @@ var renderTextLayer = (function renderTextLayerClosure() {
     },
 
     _render: function TextLayer_render(timeout) {
-      var textItems = this._textContent.items;
-      var textStyles = this._textContent.styles;
-      for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(this, textItems[i], textStyles);
+      let capability = createPromiseCapability();
+      let pump;
+      if (this._textContent === undefined) {
+        pump = () => {
+          reader.read().then(({ value, done, }) => {
+            if (done) {
+              capability.resolve();
+              return;
+            }
+            Util.appendToArray(this._textContent.items, value.items);
+            Util.extendObj(this._textContent.styles, value.styles);
+            pump();
+          });
```

Comment:
add `.catch(capability.reject);` to catch error and cancel request. Also below, properly handle the rejection by rejecting this._capability of the task.